apps/projection/src/app/service/destroy.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+
+@Injectable()
+export class DestroyService extends ReplaySubject<void> implements OnDestroy {
```

Comment:
question: why do you use a replaySubject and not a subject ?

Reply:
I used to use just a regular subject, but later I've found taiga-ui's [implementation](https://github.com/taiga-family/taiga-ui/blob/main/projects/cdk/services/destroy.service.ts
) where they use ReplaySubject and left a comment 

```
 * Why we use `ReplaySubject` instead of `Subject`?
 * Well, we’ll use ReplaySubject to emit the last message in case
 * the subscription is ended after the component is destroyed.
```
But tbh I still don't get why memory leak should happen in that case.